File: backend/app/models/review.py
"""Review Beanie document."""
from __future__ import annotations

import uuid
from typing import List, Optional

from pydantic import Field

from app.models.base import BaseDocument


class Review(BaseDocument):
    id: str = Field(default_factory=lambda: str(uuid.uuid4()))
    booking_id: str
    customer_id: str
    worker_id: str
    service_id: Optional[str] = None
    rating: float
    behaviour: int
    quality: int
    price: int
    time_rating: int
    comment: Optional[str] = None
    work_images: List[str] = Field(default_factory=list)
    recommends: bool = True

    class Settings:
        name = "reviews"

File: frontend/src/types/review.ts
export interface Review {
  _id: string;
  bookingId: string;
  customerId: string;
  customer?: { _id: string; name: string; avatar?: string };
  workerId: string;
  worker?: { _id: string; name: string; avatar?: string };
  serviceId: string;
  rating: number; // 1-5 overall
  behaviour: number;
  quality: number;
  price: number;
  time: number;
  comment: string;
  workImages: string[];
  recommends: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateReviewPayload {
  bookingId: string;
  workerId: string;
  serviceId: string;
  rating: number;
  behaviour: number;
  quality: number;
  price: number;
  time: number;
  comment: string;
  workImages: string[];
  recommends: boolean;
}

export type Feedback = Review;
